### PR TITLE
sessions for self-managed dbms docs (manual + notebook)

### PR DESCRIPTION
**Pull request by FlorentinD**
based on #712

### CODE REVIEW

File: tox.ini [hunk 1]
```diff
@@ -70,15 +70,40 @@ commands =
...
+	# for attached sessions
 	AURA_DB_ADDRESS
 	AURA_DB_USER
 	AURA_DB_PW
-	USE_DEDICATED_SESSIONS
+
+	# for standalone sessions
+	NEO4J_URI
+	NEO4J_USER
+	NEO4J_PASSWORD
```
Comment (FlorentinD) [inline tox.ini:88]:
```suggestion
```

File: doc/modules/ROOT/pages/gds-session.adoc [hunk 2]
```diff
@@ -36,6 +36,17 @@ sessions = GdsSessions(api_credentials=AuraAPICredentials(CLIENT_ID, CLIENT_SECR
 
 === Creating a GDS Session
 
+To create a GDS Session, use the `get_or_create()` method.
+It will create a new session if it does not exist, or connect to an existing one if it does.
+If the session options differ from the existing one, an error is thrown.
+
+The return value of `get_or_create()` is an `AuraGraphDataScience` object.
+It offers a similar API to the `GraphDataScience` object, but it is configured to run on a GDS Session.
+As a convention, always use the variable name `gds` for the return value of `get_or_create()`.
```
Comment (DarthMax) [inline doc/modules/ROOT/pages/gds-session.adoc:45]:
```suggestion
We recommend to name the variable holding return value of `get_or_create()` `gds`.
```

File: doc/modules/ROOT/pages/tutorials/gds-sessions.adoc [hunk 3]
```diff
@@ -65,31 +64,28 @@ sessions = GdsSessions(api_credentials=AuraAPICredentials(client_id, client_secr
...
+the manual for a full list.
 
-Lastly, we need to give our session a name. We will call ours
-`analysing-people-and-fruits`. If we had already created a session and
-want to reconnect to it, the same code is used. Doing that will not
-incur any additional costs, and will be a lot faster.
+Finally, we need to give our session a name. We will call ours
+`people-and-fruits’. If we have already created a session and want to
+reconnect to it, the same code will be used. There is no extra cost in
+doing this, and it is much faster.
```
Comment (DarthMax) [inline doc/modules/ROOT/pages/tutorials/gds-sessions.adoc:80]:
```suggestion
`people-and-fruits’. It is possible to reconnect to an existing session by calling `get_or_create` with the same session name and configuration. 
```

File: examples/gds-sessions-self-managed.ipynb [hunk 4]
```diff
@@ -0,0 +1,459 @@
...
+    "If you are using AuraDB, follow [this example](../gds-sessions)."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Prerequisites\n",
+    "\n",
+    "This notebook requires having an Neo4j instance instance available and have the GDS sessions feature enabled for your tenant. \n",
```
Comment (DarthMax) [inline examples/gds-sessions-self-managed.ipynb:45]:
```suggestion
    "This notebook requires a Neo4j instance and that the GDS session feature is enabled for your Neo4j Aura tenant. \n",
```

File: examples/gds-sessions-self-managed.ipynb [hunk 5]
```diff
@@ -0,0 +1,459 @@
...
+    "load_dotenv(\"staging_self_managed.env\", override=True)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Aura API credentials\n",
+    "\n",
+    "A GDS Session is created and accessed via the Aura API.\n",
```
Comment (DarthMax) [inline examples/gds-sessions-self-managed.ipynb:81]:
```suggestion
    "A GDS Session is managed via the Aura API.\n",
```

File: examples/gds-sessions-self-managed.ipynb [hunk 6]
```diff
@@ -0,0 +1,459 @@
...
+    "In order to use the Aura API, we need to have Aura API credentials.\n",
```
Comment (DarthMax) [inline examples/gds-sessions-self-managed.ipynb:82]:
```suggestion
    "In order to use the Aura API, we need to have [Aura API credentials](https://neo4j.com/docs/aura/platform/api/authentication/#_creating_credentials).\n",
```

File: examples/gds-sessions-self-managed.ipynb [hunk 7]
```diff
@@ -0,0 +1,459 @@
...
+    "For how to create these credentials, see the [Aura documentation](https://neo4j.com/docs/aura/platform/api/authentication/#_creating_credentials).\n",
```
Comment (DarthMax) [inline examples/gds-sessions-self-managed.ipynb:83]:
```suggestion
```

File: examples/gds-sessions-self-managed.ipynb [hunk 8]
```diff
@@ -0,0 +1,459 @@
...
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Creating a new session\n",
+    "\n",
+    "A new session is created by calling `sessions.get_or_create()`.\n",
+    "To do this, we need to identify a data source.\n",
```
Comment (DarthMax) [inline examples/gds-sessions-self-managed.ipynb:113]:
```suggestion
```

File: examples/gds-sessions-self-managed.ipynb [hunk 9]
```diff
@@ -0,0 +1,459 @@
...
+    "We assume that a self-managed Neo4j DBMS instance has been set up and is accessible.\n",
+    "We need to pass the database address, user name and password to the `DbmsConnectionInfo` class.\n",
+    "\n",
+    "We also need to specify the session size.\n",
+    "There are many possible sizes to choose from.\n",
```
Comment (DarthMax) [inline examples/gds-sessions-self-managed.ipynb:118]:
```suggestion
```

File: examples/gds-sessions-self-managed.ipynb [hunk 10]
```diff
@@ -0,0 +1,459 @@
...
+    "Please refer to the API reference documentation or the manual for a full list.\n",
+    "\n",
+    "Finally, we need to give our session a name.\n",
+    "We will call ours `people-and-fruits-sm'.\n",
+    "If we have already created a session and want to reconnect to it, the same code will be used.\n",
+    "There is no extra cost in doing this, and it is much faster."
```
Comment (DarthMax) [inline examples/gds-sessions-self-managed.ipynb:124]:
```suggestion
    " It is possible to reconnect to an existing session by calling `get_or_create` with the same session name and configuration."
```

**Review by DarthMax — APPROVED** [review]
Nice. I added a couple of suggestions

File: doc/modules/ROOT/pages/gds-session.adoc (hunk 2 above)
Comment (FlorentinD) [reply doc/modules/ROOT/pages/gds-session.adoc:45]: i like the previous one more as it does not have `get_or_create()` and `gds` right next to each other.

I always forget if we write as `we`. I think it was passive. We should have a meeting with Nicola to look over the session docs overall

Comment (DarthMax) [reply doc/modules/ROOT/pages/gds-session.adoc:45]: My problem with the first sentence is that it sounds like a requirement.

**Pull request merged**